src/hotspot/share/utilities/debug.hpp

```diff
@@ -34,14 +34,15 @@
 
 class oopDesc;
 
-// ShowRegistersOnAssert support (for now Linux only)
-#if defined(LINUX) && !defined(ZERO)
+// ShowRegistersOnAssert support (for now Linux and Windows only)
+#if (defined(LINUX) || defined(_WINDOWS)) && !defined(ZERO)
 #define CAN_SHOW_REGISTERS_ON_ASSERT
 extern char* g_assert_poison;
+extern const char* g_assert_poison_page_for_reporting;
```

Comment:
This g_assert_poison_page_for_reporting only exists to recognize assert poison crash when the assert poison itself is disarmed, right? So, during error reporting?

Makes sense but proposal for simplification:
- have two pointers initialized to the poison at arm time. E.g. "g_assert_poison" (the real one) and "g_assert_poison_copy"
- on disarm, switch g_assert_poison but leave the other unchanged
- when checking, always check against the copy - it never changes
- now you can unify the several code pieces that do `if ((si->si_signo == SIGSEGV || si->si_signo == SIGBUS) && si->si_addr == g_assert_poison_copy)` to something like `bool VMError::was_assertion_poison_crash(siginfo)`

Reply:
void disarm_assert_poison() {
  g_assert_poison_page_for_reporting = g_assert_poison;
  g_assert_poison = &g_dummy;
}

Its only initialized when the real page deactivates. The purpose here for deactivation is to not get more traps, i.e. during further reporting? 

g_assert_poison_page_for_reporting is only used in print_siginfo() to skip printing the sigInfo, only applies on Windows, I believe.